From 61b80e706aabb589efc4f835764138b7ba9f8f6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 29 Sep 2024 09:55:41 +0000
Subject: [PATCH] fix(): Handle properly if the route includes walking.

Walking routes have no stops. Use saturating_sub() to avoid integer overflow in case of zero stops.

Closes: https://gitlab.com/schmiddi-on-mobile/railway/-/issues/120
---
 src/gui/leg_item.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/leg_item.rs b/src/gui/leg_item.rs
index ec82b05..16246d2 100644
--- a/src/gui/leg_item.rs
+++ b/src/gui/leg_item.rs
@@ -138,8 +138,8 @@ pub mod imp {
                             s.iter()
                                 .filter(|l| matches!(l, rcore::IntermediateLocation::Stop(_)))
                                 .count()
-                                - 2,
-                            s.len() - 2,
+                                .saturating_sub(2),
+                            s.len().saturating_sub(2),
                         )
                     })
                     .unwrap_or_default();
-- 
GitLab

